storage: improve mvcc benchmarks under -short

RaduBerinde opened the pull request:
#### storage: improve mvcc benchmarks under -short

Many of the MVCC benchmarks have too many variants and are completely
sipped when using the `-short` flag (which is used for the weekly
microbenchmarks job). Conversely, other benchmarks are not skipped but
still have quite a few variants.

In this commit we change relevant tests to run a small set of
configurations under `-short`. We also fix a few smaller issues, for
example we remove some bogus empty acquire lock benchmarks with
`heldOtherTxn=true/heldSameTxn=true`.

Fixes #115794
Release note: None

#### storage: remove batch=true variants of BenchmarkMVCCPut

This benchmark has batching variants but the batch size is `b.N` which
can hit the 4GB batch limit.

Given that we already have a separate `MVCCBatchPut` which has
reasonable batch sizes, we are removing the batched variants of
`BenchmarkMVCCPut`.

Fixes: #115811
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115810)
<!-- Reviewable:end -->

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115810)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jbowens and @sumeerbhola)

___
*[`pkg/storage/bench_pebble_test.go` line 330 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-Nl4lkyBE7TLXyq2-Z5A:-Nl4lkyBE7TLXyq2-Z5B:bcyqlp7) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca53a6955513095f36364024b0103d87432c6fa4/pkg/storage/bench_pebble_test.go#L330)):*
> ```Go
> 	var testCases []testCase
> 	for _, batch := range []bool{false, true} {
> 		for _, numVersions := range []int{1, 10, 100} {
> ```

Note: here I am restoring the variants which were removed in #99726


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Added a commit that fixes #115811.

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-:-NlZH-px-5V3oe8D3UVd:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten, @RaduBerinde, and @sumeerbhola)

___
*[`-- commits` line 5 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-NlZGHlM57XBALFoTW6M):*
nit: "skipped when"
___
*[`pkg/storage/bench_pebble_test.go` line 330 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-Nl4lkyBE7TLXyq2-Z5A:-NlZGYQhDTzGpwqpKqkx:b5wjgsl) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca53a6955513095f36364024b0103d87432c6fa4/pkg/storage/bench_pebble_test.go#L330)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Note: here I am restoring the variants which were removed in #99726
</blockquote></details>

Oops, thanks


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
storage: improve mvcc benchmarks under -short

Many of the MVCC benchmarks have too many variants and are completely
skipped when using the `-short` flag (which is used for the weekly
microbenchmarks job). Conversely, other benchmarks are not skipped but
still have quite a few variants.

In this commit we change relevant tests to run a small set of
configurations under `-short`. We also fix a few smaller issues, for
example we remove some bogus empty acquire lock benchmarks with
`heldOtherTxn=true/heldSameTxn=true`.

Fixes #115794
Release note: None

RaduBerinde committed:
storage: remove batch=true variants of BenchmarkMVCCPut

This benchmark has batching variants but the batch size is `b.N` which
can hit the 4GB batch limit.

Given that we already have a separate `MVCCBatchPut` which has
reasonable batch sizes, we are removing the batched variants of
`BenchmarkMVCCPut`.

Fixes: #115811
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-:-NlZRRM8E5ydGLRCGxfg:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @jbowens, @nvanbenschoten, and @sumeerbhola)

___
*[`-- commits` line 5 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115810#-NlZGHlM57XBALFoTW6M):*
<details><summary><i>Previously, jbowens (Jackson Owens) wrote…</i></summary><blockquote>

nit: "skipped when"
</blockquote></details>

Haha, I like to sip my benchmarks. Fixed.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+





<!-- Sent from Reviewable.io -->
